// music.proto
syntax = "proto3";

package music;

service MusicService {
  rpc GetTrackList (GetTrackListRequest) returns (TrackListResponse);
  rpc StreamTrack (TrackRequest) returns (stream TrackChunk);
}

message Track {
  int32 id = 1;
  string title = 2;
  string artist = 3;
  string album = 4;
  string genre = 5;
  int32 duration_sec = 6;
}

message GetTrackListRequest {}
message TrackListResponse { repeated Track tracks = 1; }

message TrackRequest { int32 track_id = 1; }
message TrackChunk { bytes data = 1; }